src/site/content/en/blog/predictive-prefetching/index.md

```diff
@@ -0,0 +1,128 @@
+<iframe style="width:100%; height: 100%;position: absolute; top: 50%; left:
+50%; transform: translate(-50%,-50%);"
+src="https://www.youtube.com/embed/0jB4YWgAxUo" frameborder="0"
+allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+allowfullscreen></iframe>
+</div>
+
+## Code-Splitting for Faster Web Apps
+
+Web applications are slow, and JavaScript is among the most expensive resources that we ship to the user. Waiting for a slow web application to load can frustrate your users and decrease their conversion.
+
+![Slow web apps are stressful](guess-0.png)
+
+An efficient technique to reduce the number of bytes of JavaScript that you're transferring over the wire is to apply lazy-loading. In the context of JavaScript, you can apply lazy-loading using different techniques, for example:
+
+- Component-level code-splitting
+- Route-level code-splitting
+
+With component-level code-splitting, you can move individual components to separate JavaScript chunks. On particular events, you can load the scripts and render the components.
+
+Route-level code-splitting has similar functionality with the difference that you move entire routes into independent chunks. Once the user performs a transition from one route to another, they have to download the associated JavaScript and bootstrap the requested page. All these operations can lead to significant delays, especially on slow networks.
+
+## Prefetching JavaScript
+
+![Prefetching example](guess-1.png)
+
+Prefetching is a technique which allows you to download in the background and save into the cache resources that you think the user will need soon. The usual way to do this is by using `<link rel="prefetch">`. Two common pitfalls with prefetching are:
+
+- Prefetching too many resources and consuming a lot of data (*overfetching*)
+- Never prefetching the resources the user needs
+
+Predictive allows you to solve these problems by consuming a report for the users' navigational patterns.
+
+## Predictive Prefetching with Guess.js
+
+A JavaScript library which provides predictive prefetching functionality is [Guess.js](https://github.com/guess-js). Guess.js consumes a report from Google Analytics or another analytics provider to build a predictive model that could be used for smartly prefetching only what the user may need in the future.
+
+Guess.js has integration with Angular, Next.js, Nuxt.js, and Gatsby. To use it in your application, you should add a single line of webpack configuration specifying a [Google Analytics view ID](https://stackoverflow.com/questions/36898103/what-is-a-viewid-in-google-analytics):
+
+```js/0,5
+const { GuessPlugin } = require('guess-webpack');
+
+// ...
+plugins: [
+   // ...
+   new GuessPlugin({ GA: 'XXXXXX' })
+]
+// ...
+```
+
+If you're not using Google Analytics, you can specify a `reportProvider` and download data from your favorite service.
+
+### Integration with Angular
```

Comment:
Yep, good point. I picked Angular because I spent the most working on the integration.

I'll link to the codelabs from https://guess-js.github.io and keep the video.

Reply:
Awesome! Can you just make it an unordered list? (We save ordered lists for instructions.)